x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -248,3 +305,207 @@ const DragDropInner = React.memo(function DragDropInner(
+    defaultMessage: 'You have moved the item from position {prevPosition} to position {position}',
+    values: {
+      position,
+      prevPosition,
+    },
+  });
+
+const getKeyboardReorderMessageLifted = (position: number) =>
+  i18n.translate('xpack.lens.dragDrop.elementLifted', {
+    defaultMessage: 'You have lifted an item in position {position}',
+    values: {
+      position,
+    },
+  });
+
+const lnsLayerPanelDimensionMargin = 4;
+
+export const ReorderableDragDrop = ({
+  draggingProps,
+  dropProps,
+  children,
+  label,
+  dropTo,
+}: {
+  draggingProps: {
+    className: string;
+    draggable: Props['draggable'];
+    onDragEnd: (e: DroppableEvent) => void;
+    onDragStart: (e: DroppableEvent) => void;
+    dataTestSubj: string;
+    isReorderDragging: boolean;
+  };
+  dropProps: {
+    onDrop: (e: DroppableEvent) => void;
+    onDragOver: (e: DroppableEvent) => void;
+    onDragLeave: () => void;
+    dragging: DragContextState['dragging'];
+    droppable: DraggableProps['droppable'];
+    itemsInGroup: string[];
+    id: string;
+    isActive: boolean;
+  };
+  children: React.ReactElement;
+  label: string;
+  dropTo: DropToHandler;
+}) => {
+  const { itemsInGroup, dragging, id, droppable } = dropProps;
+  const { reorderState, setReorderState } = useContext(ReorderContext);
+
+  const { isKeyboardReorderOn, reorderedItems, draggingHeight, direction, groupId } = reorderState;
+
+  const groupLength = itemsInGroup.length;
+
+  const currentIndex = itemsInGroup.indexOf(id);
+
+  const draggingClasses = classNames(
+    draggingProps.className,
+    {
+      'lnsDragDrop-isKeyboardModeActive': isKeyboardReorderOn,
+    },
+    {
+      'lnsDragDrop-isReorderable': draggingProps.isReorderDragging,
+    }
+  );
+
+  return (
+    <div className="lnsDragDrop__reorderableContainer">
+      <EuiScreenReaderOnly showOnFocus>
+        <button
+          aria-label={`Grab ${label}`}
+          aria-describedby={`$lnsDragDrop-reorderInstructions-${groupId}`}
+          className="lnsDragDrop__keyboardHandler"
+          data-test-subj="lnsDragDrop-keyboardHandler"
+          onBlur={() => {
+            setReorderState({
+              ...reorderState,
+              isKeyboardReorderOn: false,
+              keyboardReorderMessage: '',
+            });
+          }}
+          onKeyDown={(e: React.KeyboardEvent<HTMLButtonElement>) => {
+            if (e.key === keys.ENTER || e.key === keys.SPACE) {
+              setReorderState({
+                ...reorderState,
+                isKeyboardReorderOn: !isKeyboardReorderOn,
+                keyboardReorderMessage: isKeyboardReorderOn
+                  ? ''
+                  : getKeyboardReorderMessageLifted(currentIndex + 1),
+              });
+            } else if (e.key === keys.ESCAPE) {
+              setReorderState({
+                ...reorderState,
+                isKeyboardReorderOn: false,
+                keyboardReorderMessage: '',
+              });
```

Comment:
Escape still keeps the focus on the item, is that intentional?

Reply:
Hmmm, I think so - escape allows you to escape the reordering mode, but does nothing with focus. Same for other elements on the page when they are focused.